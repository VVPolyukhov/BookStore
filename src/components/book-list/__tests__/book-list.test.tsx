import React from 'react'
import BookList, { PropsType } from '../book-list'
import { shallow } from 'enzyme'

describe('BookList:', () => {
    const props: PropsType = {
        books: [
            {
                id: 1,
                author: 'author1',
                image: 'url1',
                price: 100,
                rating: 1,
                title: 'title1'
            },
            {
                id: 2,
                author: 'author2',
                image: 'url2',
                price: 200,
                rating: 2,
                title: 'title2'
            }
        ]
    }
    const Wrapper = shallow(<BookList {...props} />)
    
    it('renders properly', () => {
        expect(Wrapper).toMatchSnapshot()
    })
    it('renders books', () => {
        expect(Wrapper.find('Connect(BookListItem)')).toHaveLength(props.books.length)
    })
})