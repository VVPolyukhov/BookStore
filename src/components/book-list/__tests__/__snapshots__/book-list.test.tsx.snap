// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList: renders properly 1`] = `
<GridColumn
  stretched={true}
  width={13}
>
  <CardGroup
    itemsPerRow={4}
  >
    <Connect(BookListItem)
      book={
        Object {
          "author": "author1",
          "id": 1,
          "image": "url1",
          "price": 100,
          "rating": 1,
          "title": "title1",
        }
      }
      key="0"
    />
    <Connect(BookListItem)
      book={
        Object {
          "author": "author2",
          "id": 2,
          "image": "url2",
          "price": 200,
          "rating": 2,
          "title": "title2",
        }
      }
      key="1"
    />
  </CardGroup>
</GridColumn>
`;
